test/put.sh

```diff
@@ -293,6 +293,33 @@ it_can_put_and_set_git_config() {
   mv ~/.gitconfig.orig ~/.gitconfig
 }
 
+it_can_put_and_force_the_push() {
+  local repo1=$(init_repo)
+
+  local src=$(mktemp -d $TMPDIR/put-src.XXXXXX)
+  local repo2=$src/repo
+  git clone $repo1 $repo2
+
+  local ref=$(make_commit $repo2)
+  make_commit $repo1
+  # cannot push to repo while it's checked out to a branch
+  git -C $repo1 checkout refs/heads/master
```

Comment:
imo it'd be clearer if this tested the more common case of force-pushing over commits - this is kind of a lesser-known behavior of local directory pushes